Userland/Libraries/LibGfx/JPEGLoader.cpp

```diff
@@ -437,9 +437,21 @@ static ErrorOr<void> build_macroblocks(JPEGLoadingContext& context, Vector<Macro
             for (u8 hfactor_i = 0; hfactor_i < scan_component.component.hsample_factor; hfactor_i++) {
                 // A.2.3 - Interleaved order
                 u32 mb_index = (vcursor + vfactor_i) * context.mblock_meta.hpadded_count + (hfactor_i + hcursor);
-                if (!context.current_scan.are_components_interleaved())
+                if (!context.current_scan.are_components_interleaved()) {
                     mb_index = vcursor * context.mblock_meta.hpadded_count + (hfactor_i + (hcursor * scan_component.component.vsample_factor) + (vfactor_i * scan_component.component.hsample_factor));
 
+                    // A.2.4 Completion of partial MCU
+                    // If the component is [and only if!] to be interleaved, the encoding process
```

Comment:
Aren't we in the `!are_components_interleaved` branch here?

Reply:
The decoder always allocates as if components are interleaved. So we skip over falsely-created macroblocks if `!are_components_interleaved`. I've updated the commit message.